cms/static/cms/js/modules/cms.base.js

```diff
@@ -261,8 +261,71 @@ var CMS = {
 
                 // ensure new settings are returned
                 return CMS.settings;
-            }
+            },
+
+            /**
+             * Modifies the url with new params and sanitises the ampersand within the url for #3404.
+             *
+             * @method makeURL
+             * @param url {String} original url
+             * @param [params] {String[]} array of `param=value` strings to update the url
+             */
+            makeURL: function makeURL(url, params) {
```

Comment:
~~because naming is hard~~
cleanURL doesn't fit, because you can add parameters
setURL i just personally don't like :D